ProcessOptimizer/space/constraints.py

```diff
@@ -121,6 +125,152 @@ def rvs(self, n_samples=1, random_state=None):
         # We draw more samples when needed so we only return n_samples of the
         # samples
         return rows[:n_samples]
+    
+    def sumequal_sampling(self, n_samples=1, random_state=None):
```

Comment:
I don't disagree per se, but the issue stems from the basic strategy for handling constraints from the ground up. All other constraint types are handled by drawing random numbers inside the original space, followed by checking if the point happened to obey the constraint(s). At its heart this is a _terribly_ inefficient strategy for finding valid points in higher dimensions, but it is very quick to implement. 

In my opinion, all constraint classes should be handled by implementing math that ensure you can _only_ draw samples that respect the constraints in the first place, which should also massively speed up the code for complex systems. However, at present I do not have the time available to open this can of worms, so I suggest that we open an issue on this, with the goal of refactoring all of `constraints.py`

Reply:
And it leads into on of my pet peeves with the package: We should refactor Space, so it becomes much modular. Individual dimensions should be a Space each, and then the whole Space should be constructed as a product of the individual Spaces. That way, constraints can simply be another Space, and just their internal logic however they want. Then, we could remove a lot of the complexity from Optimizer code, and let it reside in the Space instead.

The kernel would also reside more naturally in the space, since it is about how "far apart" points in the space is. 

I know we don't have time for it at present, but it would be SOOO nice. (Sorry for bringing it up again, I know I am rambling, no need to tell me ;-) ) So I would prefer having it as a discussion, rather than an issue. I have started that discussion here: https://github.com/novonordisk-research/ProcessOptimizer/discussions/160